src/main/takeDetection/takeDetection.ts

```diff
@@ -0,0 +1,104 @@
+import { Word, IndexRange } from 'sharedTypes';
+import {
+  InjectableTake,
+  InjectableTakeGroup,
+} from 'main/editDelete/injectTakeInfo';
+import getSimilarityScore from './sentenceSimilarity';
+
+export type Sentence = {
+  startIndex: number | null;
+  endIndex: number | null;
+  sentenceString: string;
+};
+
+export function findSentences(words: Word[]): Sentence[] {
+  let currentSentence: Sentence = {
+    startIndex: 0,
+    endIndex: null,
+    sentenceString: '',
+  };
+  const sentences: Sentence[] = [];
+  words.forEach((word, idx) => {
+    if (currentSentence.sentenceString === '') {
+      currentSentence.sentenceString = word.word;
+    } else {
+      currentSentence.sentenceString = currentSentence.sentenceString.concat(
+        ' ',
+        word.word.toString()
+      );
+    }
+
+    if (
+      word.word.includes('.') ||
```

Comment:
Getting a little bit "clever" (in a bad way) but `['.', '?', '!'].some(word.word.includes)` is a bit neater